opennms-webapp-rest/src/test/java/org/opennms/web/rest/v2/UserDefinedLinkRestServiceIT.java

```diff
@@ -0,0 +1,95 @@
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.web.rest.v2;
+
+import javax.ws.rs.core.MediaType;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.opennms.core.test.MockLogAppender;
+import org.opennms.core.test.OpenNMSJUnit4ClassRunner;
+import org.opennms.core.test.db.annotations.JUnitTemporaryDatabase;
+import org.opennms.core.test.rest.AbstractSpringJerseyRestTestCase;
+import org.opennms.netmgt.dao.DatabasePopulator;
+import org.opennms.netmgt.model.OnmsNode;
+import org.opennms.test.JUnitConfigurationEnvironment;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.web.WebAppConfiguration;
+import org.springframework.transaction.annotation.Transactional;
+
+@RunWith(OpenNMSJUnit4ClassRunner.class)
+@WebAppConfiguration
+@ContextConfiguration(locations={
+        "classpath:/META-INF/opennms/applicationContext-soa.xml",
+        "classpath:/META-INF/opennms/applicationContext-commonConfigs.xml",
+        "classpath:/META-INF/opennms/applicationContext-minimal-conf.xml",
+        "classpath:/META-INF/opennms/applicationContext-dao.xml",
+        "classpath*:/META-INF/opennms/component-service.xml",
+        "classpath*:/META-INF/opennms/component-dao.xml",
+        "classpath:/META-INF/opennms/applicationContext-databasePopulator.xml",
+        "classpath:/META-INF/opennms/mockEventIpcManager.xml",
+        "file:src/main/webapp/WEB-INF/applicationContext-svclayer.xml",
+        "file:src/main/webapp/WEB-INF/applicationContext-cxf-common.xml"
+})
+@JUnitConfigurationEnvironment
+@JUnitTemporaryDatabase
+@Transactional
+public class UserDefinedLinkRestServiceIT extends AbstractSpringJerseyRestTestCase {
+
+    public UserDefinedLinkRestServiceIT() {
+        super(CXF_REST_V2_CONTEXT_PATH);
+    }
+
+    @Autowired
+    private DatabasePopulator m_databasePopulator;
+
+    @Override
+    protected void afterServletStart() throws Exception {
+        MockLogAppender.setupLogging(true, "DEBUG");
+        m_databasePopulator.populateDatabase();
+    }
+
+    @Test
+    public void canCreateUserDefinedLinks() throws Exception {
+        final OnmsNode node1 = m_databasePopulator.getNode1();
+        final OnmsNode node2 = m_databasePopulator.getNode2();
+        String linkJson = "{\n" +
+                "      \"node-id-a\": " + node1.getId() + ",\n" +
+                "      \"component-label-a\": \"labela\",\n" +
+                "      \"node-id-z\": " + node2.getId() + ",\n" +
+                "      \"component-label-z\": \"labelb\",\n" +
+                "      \"link-id\": \"linkid\",\n" +
+                "      \"link-label\": \"my link\",\n" +
+                "      \"owner\": \"me\"\n" +
+                "}\n";
+        sendData(POST, MediaType.APPLICATION_JSON, "/userdefinedlinks", linkJson, 201);
```

Comment:
not sure if we should check if the link is actually created instead of relying on the status code? Maybe query for the created link afterwards?

Reply:
Good call.